interfaces/thirdparty/cp2k.py

```diff
@@ -340,64 +343,83 @@ def parse(key, value, indent=''):
         return inp
 
     def _parsemol(self):
-        #make lines shorter
+        # make lines shorter
         inp = self.settings.input.force_eval.subsys
-        #add cell information
+        # add cell information
         nDim = len(self.molecule.lattice)
         keys = ['A', 'B', 'C']
         periodic = ['X', 'XY', 'XYZ']
-        for iDim in range(0,nDim):
-            inp.cell[keys[iDim]] = "{:} {:} {:}".format(*self.molecule.lattice[iDim])
+        for iDim in range(0, nDim):
+            inp.cell[keys[iDim]] = "{:} {:} {:}".format(
+                *self.molecule.lattice[iDim])
         if nDim > 0:
             inp.cell.periodic = periodic[nDim-1]
 
-        #get block of: symbol coords
+        # get block of: symbol coords
         coord_sec = ""
         for atom in self.molecule:
             coord_sec += "\n"
             coord_sec += (" {:}"*4).format(atom.symbol, *atom.coords)
         inp.coord._h = coord_sec
 
     def get_runscript(self):
-        """
-        Run parallel version of Cp2k using srun.
-        """
+        """Run parallel version of Cp2k using srun."""
+        # Try to run cp2k using mpirun and otherwise srun (if available)
+        cp2k_command = self.settings.get("executable", "cp2k.popt")
+
+        # Check the executable name
+        available_executables = (
+            # Serial single core testing and debugging
+            "sdbg",
+            # Serial general single core usage
+            "sopt",
+            # Parallel (only OpenMP), single node, multi core
+            "ssmp",
+            # Parallel (only MPI) multi-node testing and debugging
+            "pdbg",
+            # Parallel (only MPI) general usage, no threads
+            "popt",
+            # parallel (MPI + OpenMP) general usage, threading might improve scalability and memory usage
+            "psmp")
+        if not any((f"cp2k.{x}" == cp2k_command.lower() for x in available_executables)):
+            msg = f"unrecognized cp2k executable: {cp2k_command}"
+            raise RuntimeError(msg)
+
         # Try to run cp2k using mpirun and otherwise srun (if available)
         command_tuple = ('mpirun', 'srun')
```

Comment:
It's probably best to use ``command = ('',)`` if one of the single-core options is used; it might cause issues otherwise.

Reply:
Right, I will change it!